rosbag2_test_common/include/rosbag2_test_common/temporary_directory_fixture.hpp

```diff
@@ -17,72 +17,26 @@
 
 #include <gmock/gmock.h>
 
-#include <cstdio>
-#include <exception>
-#include <filesystem>
-#include <fstream>
-#include <iostream>
-#include <random>
-#include <sstream>
 #include <string>
 
+#include "rcpputils/filesystem_helper.hpp"
+
 using namespace ::testing;  // NOLINT
 
 namespace rosbag2_test_common
 {
 
-/// \brief Construct a uniquely named temporary directory, in "parent", with format base_nameXXXXXX
-/// The output, if successful, is guaranteed to be a newly-created directory.
-/// The underlying implementation keeps generating paths until one that does not exist is found or
-/// until the number of iterations exceeded the maximum tries.
-/// This guarantees that there will be no existing files in the returned directory.
-/// \param[in] base_name User-specified portion of the created directory.
-/// \param[in] parent_path The parent path of the directory that will be created.
-/// \param[in] max_tries The maximum number of tries to find a unique directory (default 1000)
-/// \return A path to a newly-created directory with base_name and a 6-character unique suffix.
-/// \throws std::system_error If any OS APIs do not succeed.
-/// \throws std::runtime_error If the number of the iterations exceeds the maximum tries and
-/// a unique directory is not found.
-std::filesystem::path create_temp_directory(
-  const std::string & base_name,
-  std::filesystem::path parent_path = std::filesystem::temp_directory_path(),
-  size_t max_tries = 1000)
-{
-  // mersenne twister random generator engine seeded with the std::random_device
-  std::mt19937 random_generator(std::random_device{}());
-  std::uniform_int_distribution<> distribution(0, 999999);
-  std::filesystem::path path_to_temp_dir;
-  constexpr size_t kSuffixLength = 7;  // 6 chars + 1 null terminator
-  char random_suffix_str[kSuffixLength];
-  size_t current_iteration = 0;
-  while (true) {
-    snprintf(random_suffix_str, kSuffixLength, "%06x", distribution(random_generator));
-    const std::string random_dir_name = base_name + random_suffix_str;
-    path_to_temp_dir = parent_path / random_dir_name;
-    // true if the directory was newly created.
-    if (std::filesystem::create_directory(path_to_temp_dir)) {
-      break;
-    }
-    if (current_iteration == max_tries) {
-      throw std::runtime_error(
-              "Exceeded maximum allowed iterations to find non-existing directory");
-    }
-    current_iteration++;
-  }
-  return path_to_temp_dir;
-}
-
 class TemporaryDirectoryFixture : public Test
 {
 public:
   TemporaryDirectoryFixture()
   {
-    temporary_dir_path_ = create_temp_directory("tmp_test_dir_").string();
+    temporary_dir_path_ = rcpputils::fs::create_temporary_directory("tmp_test_dir_").string();
   }
 
   ~TemporaryDirectoryFixture() override
   {
-    std::filesystem::remove_all(std::filesystem::path(temporary_dir_path_));
+    rcpputils::fs::remove_all(rcpputils::fs::path(temporary_dir_path_));
```

Comment:
@ahcorde It will be better to use `std::filesystem::remove_all(path)` for deletion. 
Eventually, we want to deprecate and remove `rcpputils::fs::remove_all`